pkg/auth/identity.go

```diff
@@ -0,0 +1,70 @@
+package auth
+
+import (
+	"context"
+
+	authapi "k8s.io/api/authorization/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Authorized map[string]bool
+
+type Identity struct {
+	Token   string
+	RestCfg *rest.Config
+	client  k8sclient.Client
+}
+
+func (r *Identity) HasRead(namespaces []string) (Authorized, error) {
```

Comment:
Seems like this should only raise an error when it fails.
- General client API fail.
- Authentication failed.

The caller can use the authorized list to determine if any of the namespaces are not authorized.

Same for `HasMigrate()`.

Reply:
Perhaps a separate `Authenticate()` that does a `TokenReview` and report that as a separate token authentication condition.

Reply:
@jortel I'm a little confused here, this func currently only returns an error for those two cases you described. it returns an error if:
1. building the client fails
2. Running the client SAR check fails to execute (this is when the authentication fails)

For this reason I felt that it was superfluous to even use `TokenReview` and/or to have a separate `Authenticate()` func. Thoughts?

Reply:
I was wrong, I ended up doing this.